test/bandit/http1/request_test.exs

```diff
@@ -252,6 +257,23 @@ defmodule HTTP1RequestTest do
       raise "boom"
     end
 
+    test "returns user-defined date header instead of bandits", context do
+      {:ok, response} =
+        Finch.build(:get, context[:base] <> "/date_header")
+        |> Finch.request(context[:finch_name])
+
+      assert response.status == 200
+
+      assert List.keyfind(response.headers, "date", 0) |> elem(1) ==
+               "Tue, 27 Sep 2022 07:17:32 GMT"
+    end
+
+    def date_header(conn) do
+      conn
+      |> put_resp_header("DaTe", "Tue, 27 Sep 2022 07:17:32 GMT")
```

Comment:
Based on comments on https://hexdocs.pm/plug/Plug.Conn.html#put_resp_header/3 I think we can rely on `date` being lowercase, especially if it gains some performance in a hot path.